backend/app/adapter/google/identityprovider_integration_test.go

```diff
@@ -0,0 +1,137 @@
+// +build integration all
+
+package google
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/short-d/app/mdtest"
+)
+
+func TestIdentityProvider_GetAuthorizationURL(t *testing.T) {
+	t.Parallel()
+	httpRequest := mdtest.NewHTTPRequestFake(
+		func(req *http.Request) (response *http.Response, e error) {
+			return nil, nil
+		})
+	clientID := "id_12345"
+	clientSecret := "client_secret"
+	redirectURI := "http://localhost/oauth/google/sign-in/callback"
+	identityProvider := NewIdentityProvider(httpRequest, clientID, clientSecret, redirectURI)
+
+	url := identityProvider.GetAuthorizationURL()
+
+	mdtest.Equal(t, "https://accounts.google.com/o/oauth2/v2/auth?client_id=id_12345&include_granted_scopes=true&"+
```

Comment:
Let's parse the url using [this](https://golang.org/pkg/net/url/#Parse) function so that we can verify the correctness of result in order independent mannar.